wordpress/mod.ts

```diff
@@ -32,178 +126,158 @@ export default function (userOptions?: Partial<Options>) {
+    let pageIndex = 1;
     let counter = 0;
 
     while (limit > counter) {
-      const per_page = Math.min(limit - counter, max_per_page);
-      counter += per_page;
+      const pageSize = Math.min(limit - counter, this.#maxPerPage);
+      counter += pageSize;
+
       const posts = await this.#fetch<T>(
-        `${path}?per_page=${per_page}&page=${page++}`,
+        apiPath,
+        [
+          ["page", `${pageIndex++}`],
+          ["per_page", `${pageSize}`],
+        ],
       );
 
       for (const post of posts) {
         yield post;
       }
 
-      if (posts.length < max_per_page) {
+      if (posts.length < this.#maxPerPage) {
         break;
       }
     }
   }
 
-  async #fetch<T>(path: string): Promise<T[]> {
-    const url = new URL(path, this.#api_endpoint);
+  async #fetch<T>(apiPath: string, query: QueryParams = []): Promise<T[]> {
+    const url = this.#craftUrl(apiPath, query);
     const cacheKey = url.toString();
 
     if (this.#cache.has(cacheKey)) {
-      return this.#cache.get(cacheKey);
+      return this.#cache.get(cacheKey) as T[];
+    }
+
+    const headers = this.#pass
+      ? { Authorization: `Basic ${this.#pass}` }
+      : undefined;
+    const res = await fetch(url, { headers });
+
+    if (!res.ok) {
+      throw new Error(
+        `Unable to fetch the data: ${res.status}»${res.statusText}`,
+      );
     }
 
-    const res = await fetch(url);
-    const data = await res.json() as T[];
+    const data = await res.json();
     this.#cache.set(cacheKey, data);
     return data;
   }
-}
 
-function postToData(post: WP_REST_API_Post): Partial<PageData> {
-  return {
-    id: post.id,
-    url: new URL(post.link).pathname,
-    slug: post.slug,
-    date: new Date(post.date),
-    title: post.title.rendered,
-    content: post.content.rendered,
-    excerpt: post.excerpt.rendered,
-    category_id: post.categories,
-    tag_id: post.tags,
-    sticky: post.sticky,
-    author_id: post.author,
-    type: "post",
-  };
-}
+  #craftUrl(apiPath: string, query_params: QueryParams = []): URL {
+    const params = new URLSearchParams([
+      ...query_params,
+      [WordPressAPI.#apiRoute, apiPath],
+    ]);
 
-function pageToData(post: WP_REST_API_Post): Partial<PageData> {
-  return {
-    id: post.id,
-    url: new URL(post.link).pathname,
-    slug: post.slug,
-    date: new Date(post.date),
-    title: post.title.rendered,
-    content: post.content.rendered,
-    excerpt: post.excerpt.rendered,
-    author_id: post.author,
-    type: "page",
-  };
-}
+    return new URL("/index.php?" + params.toString(), this.#base);
+  }
 
-function termToData(term: WP_REST_API_Term): Partial<PageData> {
-  return {
-    id: term.id,
-    url: new URL(term.link).pathname,
-    slug: term.slug,
-    title: term.name,
-    name: term.name,
-    count: term.count,
-    type: term.taxonomy === "category" ? "category" : "tag",
-  };
+  addTransformer(transform: Transform) {
+    for (const collectionName in transform) {
```

Comment:
I think this is cleaner:

```suggestion
    for (const [name, transformer] of Object.entries(transform)) {
```
So you can remove the other lines below

Reply:
Nice trick!